### PR TITLE
Fix/5.9.x missing with parser token

### LINKED ISSUE #563: Issue parsing « WITH PARSER `ngram` »
It seems the parser has an issue with fulltext index with ngram, which throws an exception. I've seen it on phpmyadmin: foreign keys (that are defined after the fulltext) are missing on the table's relations page.

Simple example with:

```sql
CREATE TABLE `mytable` (
  `column_one` smallint unsigned DEFAULT NULL,
  FULLTEXT KEY `fulltext_index` (`column_one`) /*!50100 WITH PARSER `ngram` */,
  CONSTRAINT `my_constraint_1` FOREIGN KEY (`column_one`) REFERENCES `other_table` (`column_two`) ON DELETE RESTRICT ON UPDATE RESTRICT
)
```

Exception: `A comma or a closing bracket was expected.`
Exception debug for `->token`:
```
public 'token' => 
            object(PhpMyAdmin\SqlParser\Token)[1299]
              public 'token' => string 'WITH' (length=4)
              public 'value' => string 'WITH' (length=4)
              public 'keyword' => string 'WITH' (length=4)
              public 'type' => int 1
              public 'flags' => int 3
              public 'position' => int 128
```

### PR DESCRIPTION
**Pull request by niconoe-**
Resolves #563

### CODE REVIEW

File: tests/data/parser/parseCreateTableWithParser.in [hunk 1]
```diff
@@ -0,0 +1,5 @@
+CREATE TABLE `mytable` (
+  `column_one` smallint unsigned DEFAULT NULL,
+  FULLTEXT KEY `fulltext_index` (`column_one`) /*!50100 WITH PARSER `ngram` */,
+  CONSTRAINT `my_constraint_1` FOREIGN KEY (`column_one`) REFERENCES `other_table` (`column_two`) ON DELETE RESTRICT ON UPDATE RESTRICT
+)
```
Comment (iifawzi) [inline tests/data/parser/parseCreateTableWithParser.in:5]: This query is invalid, full text indexes can't be created on int fields, can we replace it with varchar instead?

**Comment by iifawzi** [comment]
Super 🚀 , thank you!

File: tests/data/parser/parseCreateTableWithParser.in (hunk 1 above)
Comment (niconoe-) [reply tests/data/parser/parseCreateTableWithParser.in:5]: Done on both PRs.

**Review by iifawzi — APPROVED** [review]
LGTM

**Comment by niconoe-** [comment]
@MauricioFauth As you merged #575, I think you can also merge this one.

I know there are 2 failing chekcs but these are because of benchmarks not being properly configured. That's fixed on 5.10.x branch, and out of this scope.

Thanks :slightly_smiling_face:

**Pull request closed without merging**